doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -718,6 +718,33 @@ the toplevel, and using them in source files is discouraged.
    the :ref:`command line option <command-line-options>` :n:`-I @string` or by the command :cmd:`Add
    ML Path` `@string` (cf. :cmd:`Declare ML Module`).
 
+.. _extra_dependencies:
+
+Extra Dependencies
+------------------
+
+Dependencies on external files, i.e. non ``.v`` files, can be declared as
+follows:
+
+.. cmd:: From @dirpath Extra Dependency @string {? as @ident }
+   :name: From … Dependency
+
+   Adds an additional dependency of the current `.v`  file on an external file.  This
+   information is included in the ``coqdep`` tool generated list of dependencies.
+   The file name :n:`@string` must exist relative to only one of the top directories
+   associated with :n:`@dirpath`.  :n:`@string` can include directory separators
+   (``/``) to select a file in a subdirectory.
+   Path elements in :n:`@string` must be valid Coq identifiers, e.g. they cannot
+   contain characters such as ``-`` or ``,``.  See :ref:`lexical-conventions`.
+
+When :n:`@ident` is provided, that name can be used by OCaml code, typically
+in a plugin, to access the full path of the external file via the API
+``ComExtraDeps.query_extra_dep``.
+
+This command has been available through the :cmd:`Comments` command,
```

Comment:
Why "has been" rather than "is also"?

Another remark is that it's a bit surprising to me that the documentation of this command is not a little more specific about the use case.

Anyway, these are very minor comments so take them into account or not as you wish.

Reply:
I didn't mean put an example. Just mention that things like Elpi exist.

Reply:
> Why "has been" rather than "is also"?

My suggestion, based on the understanding (from the original wording) that the Comments form is deprecated.  If that's incorrect, "is also" may be more accurate.

Reply:
And AFAIK was undocumented.

Reply:
@jfehrle Comments is not new, since 8.4.5 (I can't test further back) it has worked. You can do for example `Comments asdf asdf asdf.` Enrico's trick here is to use that for backwards compatibility since it will be ignored by previous versions but can be picked up on master until developments can catch up.

Reply:
The `Comments` command is not new.  Associating an action with `Comments From ... Extra Dependency` is deprecated (the "Comments form" I referred to) in favor of the new command `From ... Extra Dependency`.  Right?

Reply:
Right, currently in master they do the exact same thing, but older versions of Coq will only be happy about (not understand tho) the Comments version.

Reply:
You mean "newer versions", right?

Reply:
Thanks for clarifying. Still I don't know how to improve this text if needed.

Reply:
> Associating an action with `Comments From ... Extra Dependency` is deprecated

It's both new and deprecated at the same time. Previously, one could write `Comments From ... Extra Dependency` but it would not be associated with any action.

Reply:
> It's both new and deprecated at the same time.

A form of planned obsolescence??  I still don't get it, but since is already merged, it matters little.

Reply:
So Enrico introduced a new command `From ... Extra Dependency`. If you put this in a Coq file, master will now compile since this PR has merged. But if you were to put this in Coq < 8.15, Coq would reject the command since it doesn't know what it is. This `Comments` trick allows you to write such a command in a backwards compatible way, what Enrico did was to intercept the parsing of `Comments From ... Extra Dependency` so that it did the same thing as `From ... Extra Dependency`. Now Coq projects can put `Comments From ... Extra Dependency` and keep backwards compatibility. It is immediately deprecated since in the future we do plan to get rid of this particular interpretation of `Comments`, hence we are encouraging people to use the new `From ... Extra Dependency ` if possible.